term-utils/scriptreplay.c

```diff
@@ -308,9 +320,64 @@ main(int argc, char *argv[])
 		replay_set_delay_max(setup, &maxdelay);
 	replay_set_delay_min(setup, &mindelay);
 
-	isterm = setterm(&saved);
+	isterm = setterm(&saved, &saved_flag);
 
 	do {
+		switch (fgetwc(stdin)) {
+			case ' ':
+				replay_toggle_pause(setup);
+				break;
+			case '\033':
+				wchar_t firstChar = fgetwc(stdin);
+				if (firstChar == '[') {
+					wchar_t secondChar = fgetwc(stdin);
+
+					if (secondChar == 'A') { // Up arrow
+						divi += 0.1;
+						replay_set_delay_div(setup, divi);
+					} else if (secondChar == 'B') { // Down arrow
+						divi -= 0.1;
+						if (divi < 0.1)
+							divi = 0.1;
+						replay_set_delay_div(setup, divi);
+					} else if (secondChar == 'C') { // Right arrow
+						rc = replay_emit_step_data(setup, step, STDOUT_FILENO);
+						if (!rc) {
+							rc = replay_get_next_step(setup, streams, &step);
+							if (!rc) {
+								struct timeval *delay = replay_step_get_delay(step);
+								if (delay && timerisset(delay))
+									stepDelay = *delay;
+							}
+						}
+					}
+				}
+				break;
+		}
+		if (rc)
+			break;
+
+		if (replay_get_is_paused(setup))
+		{
```

Comment:
Please follow the current coding style. It means "if ( ) {" rather than "if ( )\n{".
https://www.kernel.org/doc/html/v4.10/process/coding-style.html#placing-braces-and-spaces

We also do not use "//" for comments :-)

Reply:
Fixed with https://github.com/util-linux/util-linux/pull/3015/commits/0bfd75dbf413be85049e0c89551f208cfa64a2ea